src/Raven.Server/ServerWide/Commands/ModifyDatabaseWatchersCommand.cs

```diff
@@ -64,14 +65,22 @@ public override string UpdateDatabaseRecord(DatabaseRecord record, long etag)
             if (Watcher == null)
                 return null;
 
-            var newTaskKey = Watcher.GetTaskKey().ToString();
-            if (Watcher.CurrentTaskId != null && Watcher.CurrentTaskId != newTaskKey)
+            var newTaskId = (Watcher.Database + Watcher.Url).ToLower();
```

Comment:
but if i don't keep a task id, i'm facing a problem: when a watcher's db or url changes, i don't know what it was before the change and therefore i don't know which watcher to remove from the list.
I also won't be able to know if it's new or modified.